tests/integration/TestCase.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * (c) Toby Zerner <toby.zerner@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Flarum\Tests\integration;
+
+use Flarum\Foundation\InstalledSite;
+use Illuminate\Database\ConnectionInterface;
+
+abstract class TestCase extends \PHPUnit\Framework\TestCase
+{
+    public function setUp()
+    {
+        parent::setUp();
+
+        // Boot the Flarum app
+        $this->app();
+    }
+
+    protected $app;
+
+    /**
+     * @return \Flarum\Foundation\InstalledApp
+     */
+    protected function app()
+    {
+        if (! is_null($this->app)) {
+            return $this->app;
+        }
+
+        $site = new InstalledSite(
+            [
+                'base' => __DIR__.'/tmp',
+                'public' => __DIR__.'/tmp/public',
+                'storage' => __DIR__.'/tmp/storage',
+            ],
+            include __DIR__.'/tmp/config.php'
+        );
+
+        return $this->app = $site->bootApp();
+    }
+
+    protected $database;
+
+    protected function database(): ConnectionInterface
+    {
+        if (is_null($this->database)) {
+            $this->database = $this->app()->getContainer()->make(
+                ConnectionInterface::class
+            );
+        }
+
+        return $this->database;
+    }
+
+    protected function prepareDatabase(array $tableData)
+    {
+        // We temporarily disable foreign key checks to simplify this process.
+        $this->database()->statement('SET FOREIGN_KEY_CHECKS = 0');
```

Comment:
```suggestion
        $this->database()->getSchemaBuilder()->disableForeignKeyConstraints();
```

That might be better cross-database-driver compatible ;)